### DISCUSSION TITLE
Issues with port forwarding in Codespaces + VSCode Desktop

### DISCUSSION THREAD

**Discussion by alessandro-newzoo** [community/community#69916] [Codespaces]
### Select Topic Area

Bug

### Body

_Possibly related to https://github.com/orgs/community/discussions/68854 which was allegedly fixed.
I've already sent a [ticket](https://support.github.com/ticket/personal/0/2374329) for this but I'd like to post it here too in case someone else is facing the same problem as it's quite serious._

----

I'm having issues concerning forwarding Codespace ports in VS Code Desktop. 
They are not happening on VS Code Web.

## The problems
1. VSCode will map ports to the wrong/old `preview.app.github.dev` domain (which contains `preview`).
   - the right URL should be `https://improved-journey-pp9rjqpr5wj264x9-3000.app.github.dev`
   - but VS Code is mapping it to `https://improved-journey-pp9rjqpr5wj264x9-3000.preview.app.github.dev`
2. If the ports are set to **Private**, they will be mapped to `127.0.0.1` instead of the URL. Once I set them to public, they will switch to the URL. 

![image.png](https://github.zendesk.com/attachments/token/yQXY4Vzie19Mjp3UCu8KDWZQo/?name=image.png)

I've asked two friends on Discord to reproduce the steps and they bumped into the exact same issue.

## Steps to reproduce
1. Create a new codespace with only this `devcontainer.json` in the repo
```
{
  "image": "mcr.microsoft.com/devcontainers/universal:2",
  "forwardPorts": [3000]
}

```
2. Launch it in VS Code Desktop

## My environment
```
MacBook Pro M1
13.6 (22G120)

VS Code 1.83.1 
VS Code 1.84.0-insider
No extensions installed except for the 'Codespaces' and 'Dev Containers' extensions.
```

**Codespace name**
`improved-journey-pp9rjqpr5wj264x9`

**Public Repo**
https://github.com/newzoo-com/test/tree/main

----

Thank you,
Alessandro

**Comment by richardberube** [comment]
As of today, Feb 21st, 2024, this issue is still going on. Port forwarding works correctly on the web, but when you port forward with the vscode extension, the link still contain `preview` . If you remove it manualy it work.. but it still generate the wrong link.

Issue for `private` are still wrongly mapped to 127.0.0.1 instead of `app.github.dev`
